coderd/audit/request.go

```diff
@@ -150,11 +158,19 @@ func InitRequest[T Auditable](w http.ResponseWriter, p *RequestParams) (*Request
 			p.AdditionalFields = json.RawMessage("{}")
 		}
 
+		var userID uuid.UUID
+		key, ok := httpmw.APIKeyOptional(p.Request)
+		if ok {
+			userID = key.UserID
+		} else {
+			userID = req.UserID
+		}
```

Comment:
:+1: 

If we have a "Anonymous user" this is probably where we would do that is `req.UserID.IsNil()`

Reply:
I think anonymous users (with UserID of `uuid.Nil`) are handled in this conditional in the `else` clause - `req.UserID` will always be nil if not explicitly passed in with a value.
Let me know if I'm misunderstanding what you're saying!